pkg/bucketaws/setup.go

```diff
@@ -18,64 +18,131 @@ package s3setup
 import (
 	"fmt"
 	"os"
+	"strings"
+	"time"
+
+	files "github.com/redhatcre/syncron/utils"
+
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 )
 
+func ProcessDate(fromDate time.Time) []string {
+
+	// This function formats all dates from provided date
+	// until current date and returns a string formatted
+	// for s3 keys.
+
+	start := fromDate
+	end := time.Now()
+	var dates []string
+	for d := start; !d.After(end); d = d.AddDate(0, 0, 1) {
+		year, month, day := d.Date()
+		dates = append(
+			dates,
+			fmt.Sprintf(
+				"created_year=%d/created_month=%d/created_day=%d",
+				year, month, day))
+	}
+	return dates
+}
+
 func ConfigRead() error {
+
```

Comment:
I think that the return type of `ConfigRead` should be adjusted, since it no longer returns any error, it either succeds and returns nil, or fails and exits

Reply:
Good point, I will adjust it